kvbc/src/replica_resources.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 VMware, all rights reserved
+
+#include "replica_resources.h"
+#include "Logger.hpp"
+#include <thread>
+
+using namespace concord::performance;
+using namespace std::chrono_literals;
+
+void ReplicaResourceEntity::addMeasurement(const ISystemResourceEntity::measurement &m) {
+  if (is_stopped) {
+    return;
+  }
+  // don't wait for the lock if not available, it's ok to loose small amount of measurements
+  if (!lock(adding_measurements_thread, dont_wait)) return;
```

Comment:
you also need to unlock here, otherwise, the other thread won't have the chance to "grab" the lock until this thread will mark itself as the victim again